app/controller.go

```diff
@@ -89,6 +98,53 @@ func (fn HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+const headerContentType = "Content-Type"
+
+type HostForwarder struct {
+	AddressResolver HostAddressResolver
+	Client          *http.Client
+}
+
+func (f *HostForwarder) Handler() HTTPHandler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		zone := vars["zone"]
+		if zone == "" {
+			return NewBadRequestError("Invalid url, missing zone value", nil)
+		}
+		host := vars["host"]
+		if host == "" {
+			return NewBadRequestError("Invalid url, missing host value", nil)
+		}
+		split := strings.SplitN(r.URL.String(), "hosts/"+host+"/", 2)
+		if len(split) != 2 {
+			return NewBadRequestError("Invalid url, missing host resource", nil)
+		}
+		addr, err := f.AddressResolver.GetHostAddr(zone, host)
+		if err != nil {
+			return err
+		}
+		url := fmt.Sprintf("%s/%s", addr, split[1])
+		req, err := http.NewRequest(r.Method, url, r.Body)
+		if err != nil {
+			return nil
+		}
+		resp, err := f.Client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		b, err := ioutil.ReadAll(resp.Body)
```

Comment:
You could avoid most of this boilerplate by using an instance of [ReverseProxy](https://pkg.go.dev/net/http/httputil#ReverseProxy). I think you'll only need to provide an implementation for `Director` and leave everything else as the default. Or you could use `NewSingleHostReverseProxy` as is done in `signalingserver.go`.

Reply:
definitely better, created the `ReverseProxy` instance passing the `Director` to avoid modifying the path of the original request.